src/Orleans.Core/Core/ClientBuilderExtensions.cs

```diff
@@ -173,5 +177,46 @@ public static IClientBuilder UseStaticGatewayListProvider(this IClientBuilder bu
             return builder.ConfigureServices(collection =>
                 collection.UseStaticGatewayListProvider(configureOptions));
         }
+
+        /// <summary>
+        /// Returns the <see cref="ApplicationPartManager"/> for this builder.
+        /// </summary>
+        /// <param name="builder">The builder.</param>
+        /// <returns>The <see cref="ApplicationPartManager"/> for this builder.</returns>
+        public static ApplicationPartManager GetApplicationPartManager(this IClientBuilder builder) => ApplicationPartManagerExtensions.GetApplicationPartManager(builder.Properties);
+
+        /// <summary>
+        /// Adds an <see cref="IApplicationPart"/> to the <see cref="ApplicationPartManager"/>.
+        /// </summary>
+        /// <param name="builder">The builder.</param>
+        /// <param name="assembly">The assembly.</param>
+        /// <returns>The builder.</returns>
+        public static IClientBuilder AddApplicationPart(this IClientBuilder builder, Assembly assembly)
+        {
+            builder.GetApplicationPartManager().AddApplicationPart(new AssemblyPart(assembly));
+            return builder;
+        }
+
+        /// <summary>
+        /// Configures the <see cref="ApplicationPartManager"/> for this builder.
+        /// </summary>
+        /// <param name="builder">The builder.</param>
+        /// <param name="configure">The configuration delegate.</param>
+        /// <returns>The builder.</returns>
+        public static IClientBuilder ConfigureApplicationPartManager(this IClientBuilder builder, Action<IApplicationPartManager> configure)
```

Comment:
How about dropping *Manager* and just have `ConfigureApplicationParts(...)`? It is using the manager under the hoods, but the user is configuring which parts the app will use (and of course it's less verbose)